### ISSUE TITLE
API documentation

### ISSUE THREAD

**Issue by hoyes** [camdram/camdram#423]
https://symfony.com/doc/current/bundles/NelmioApiDocBundle/index.html looks widely used, and automatically generates documentation from annotations on the REST methods themselves.

The latest version requires PHP 7 though, and there look to be some big changes in the format of the annotations between 2.x and 3.x so might be better to wait until we're running PHP 7 everywhere before implementing.

**Comment by CHTJonas** [comment]
Waiting on #426

**Comment by CHTJonas** [comment]
Had a look at integrating NelmioApiDocBundle and it exposes quite a lot of paths that look like they might be vulnerable to the issues discussed in #533 so have placed on hold for the time being. There are ways to exclude endpoints but I'd rather fix the issues at source rather than run with a mountain of excludes.

**Comment by hoyes** [comment]
A set of tests for a useful subset of the API that we could maybe adapt and incorporate into Camdram's test suite: https://github.com/CHTJonas/camdram-ruby/tree/master/test (with thanks to @CHTJonas)

**Comment by GKFX** [comment]
As a more manual approach I've started an API reference at https://camdram.github.io/api: the Camdram API is relatively small and static so it may be simpler just to write up by hand.
